src/toRedactor.tsx

```diff
@@ -118,6 +120,45 @@ const ELEMENT_TYPES: IJsonToHtmlElementTags = {
     } else if (extraAttrs?.displayType === 'asset') {
       return `<figure${attrs}>${child}</figure>`
     }
+     
+    else if (extraAttrs?.displayType === "display") {
+      const anchor = jsonBlock?.["attrs"]?.["link"];
+
+      const caption = jsonBlock?.["attrs"]?.["asset-caption"];
+      const position = jsonBlock?.["attrs"]?.["position"];
+      const inline = jsonBlock?.["attrs"]?.["inline"]
+      
+      attrs = ` src="${jsonBlock?.["attrs"]?.["asset-link"]}"` + attrs;
+      let img = `<img${attrs}/>`;
+
+      if (anchor) {
+        const target = jsonBlock?.["attrs"]?.["target"];
+        let anchorAttrs = `href="${anchor}"`;
+        if (target) {
+          anchorAttrs = `${anchorAttrs} target="${target}"`;
+        }
+        img = `<a ${anchorAttrs}>${img}</a>`;
+      }
+
+      if (caption || (position && position !== "none")) {
+        const figcaption = `<figcaption style="text-align:center">${caption}</figcaption>`;
```

Comment:
what if the caption is empty here? we should avoid figcaption if caption is not there...

Reply:
also i had one more query...we have to make sure that if the child has a figcaption/caption then it must have figure parent at some level...

Reply:
Yes, the figure tag will be prepended if the image is not inline, and if the image is inline only if the image has alignment or caption